src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs

```diff
@@ -1002,6 +1002,84 @@ class Z
 optionsCallback: options => options[0].Value = true);
         }
 
+        [WorkItem(41428, "https://github.com/dotnet/roslyn/issues/41428")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructorFromMembers)]
+        public async Task TestAddNullChecksWithNullableReferenceType()
+        {
+            await TestWithPickMembersDialogAsync(
+@"
+using System;
+using System.Collections.Generic;
+#nullable enable
+
+class Z
+{
+    int a;
+    string b;
+    string? c;
+    [||]
+}",
+@"
+using System;
+using System.Collections.Generic;
+#nullable enable
+
+class Z
+{
+    int a;
+    string b;
+    string? c;
+
+    public Z(int a, string b, string? c{|Navigation:)|}
+    {
+        this.a = a;
+        this.b = b ?? throw new ArgumentNullException(nameof(b));
+        this.c = c;
+    }
+}",
+chosenSymbols: new string[] { "a", "b", "c" },
+optionsCallback: options => options[0].Value = true);
+        }
+
+        [WorkItem(41428, "https://github.com/dotnet/roslyn/issues/41428")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructorFromMembers)]
+        public async Task TestAddNullChecksWithNullableReferenceTypeForGenerics()
+        {
+            await TestWithPickMembersDialogAsync(
+@"
+using System;
+using System.Collections.Generic;
+#nullable enable
+
+class Z<T> where T : class
+{
+    int a;
+    string b;
+    T? c;
+    [||]
+}",
+@"
+using System;
+using System.Collections.Generic;
+#nullable enable
+
+class Z<T> where T : class
+{
+    int a;
+    string b;
+    T? c;
+
+    public Z(int a, string b, T c{|Navigation:)|}
```

Comment:
looks like the `T?` was lost here.  intentional? a remaining bug?

Reply:
Remaining bug. Was going to open a separate issue for it, since this one is about the generated null checks.

Should I look it in the scope of this issue? I'm up to fix it.

Reply:
if it's cheap to fix, go for it :)

otehrwise, we can open a new issue.  Up to you! :)

Reply:
I'm working on it, but my local environment has broken down for some reason, so this might take a little while. If it's better to merge this PR instead of waiting for me to fix the other issue, I'll open a new issue for this.